src/node_wasm_web_api.cc

```diff
@@ -0,0 +1,161 @@
+#include "node_wasm_web_api.h"
+
+#include "env-inl.h"
+#include "node_errors.h"
+
+namespace node {
+namespace wasm_web_api {
+
+v8::Local<v8::Function> WasmStreamingObject::Initialize(Environment* env) {
+  v8::Local<v8::Function> templ = env->wasm_streaming_object_constructor();
+  if (!templ.IsEmpty()) {
+    return templ;
+  }
+
+  v8::Local<v8::FunctionTemplate> t = env->NewFunctionTemplate(New);
+  t->Inherit(BaseObject::GetConstructorTemplate(env));
+  t->InstanceTemplate()->SetInternalFieldCount(
+      WasmStreamingObject::kInternalFieldCount);
+
+  env->SetProtoMethod(t, "push", Push);
+  env->SetProtoMethod(t, "finish", Finish);
+  env->SetProtoMethod(t, "abort", Abort);
+
+  auto function = t->GetFunction(env->context()).ToLocalChecked();
+  env->set_wasm_streaming_object_constructor(function);
+  return function;
+}
+
+void WasmStreamingObject::RegisterExternalReferences(
+    ExternalReferenceRegistry* registry) {
+  registry->Register(Push);
+  registry->Register(Finish);
+  registry->Register(Abort);
+}
+
+v8::MaybeLocal<v8::Object> WasmStreamingObject::Create(
+    Environment* env, std::shared_ptr<v8::WasmStreaming> streaming) {
+  v8::Local<v8::Function> ctor = Initialize(env);
+  v8::Local<v8::Object> obj;
+  if (!ctor->NewInstance(env->context(), 0, nullptr).ToLocal(&obj)) {
+    return v8::MaybeLocal<v8::Object>();
+  }
+
+  CHECK(streaming);
+
+  WasmStreamingObject* ptr = Unwrap<WasmStreamingObject>(obj);
+  CHECK_NOT_NULL(ptr);
+  ptr->streaming_ = streaming;
+  return obj;
+}
+
+void WasmStreamingObject::New(const v8::FunctionCallbackInfo<v8::Value>& args) {
+  CHECK(args.IsConstructCall());
+  Environment* env = Environment::GetCurrent(args);
+  new WasmStreamingObject(env, args.This());
+}
+
+void WasmStreamingObject::Push(
```

Comment:
unless i am mistaken, these methods do not allocate. would you be up for adding fast call wrappers? if not i can maybe add them in a followup pr.

Reply:
Will try that tomorrow. So far, every time I tried using fast calls, I can into some V8 limitation, but `Push` might be a good candidate for once.

Reply:
On second thought, given how web streams are performing so far and how slow everything is, that seems like premature optimization. I might create a follow-up PR to add that or I might push that as a separate commit later, but I'd like to get this landed rather sooner than later and not make it more complicated to review.